From 48912910f6449a49f63f9030b1607e6f372faf26 Mon Sep 17 00:00:00 2001
From: Tonu Zalpure <tonux.zalpure@intel.com>
Date: Thu, 19 Feb 2015 20:23:26 +0530
Subject: [PATCH] EM/XPWRFG: Added null character at the end of battery model
 name

Battery model name was not terminated with null character
which caused garbage character printing in battery sysfs.
Fixed this issue by adding null character at the end of string.

Change-Id: Iacd116fe43746e28a079377d388dc06efbfb7f76
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1050
Signed-off-by: Tonu Zalpure <tonux.zalpure@intel.com>
---
 drivers/mfd/intel_soc_pmic_dc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index 239f3fb..e15fe0b 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -543,9 +543,11 @@ static void dc_xpwr_fg_pdata(void)
 		|| (!pse_mod_prof))
 		snprintf(pdata.battid, (BATTID_STR_LEN + 1),
 			"%s", BATTID_UNKNOWN);
-	else
+	else {
 		memcpy(pdata.battid, pse_mod_prof->batt_id,
 				strlen(pse_mod_prof->batt_id));
+		pdata.battid[BATTID_STR_LEN] = '\0';
+	}
 
 	platform_set_battery_data(&pdata, &ps_batt_chrg_prof);
 	pse_mod_prof = (struct ps_pse_mod_prof *)
-- 
1.9.1

